scio-bigquery/src/main/scala/com/spotify/scio/bigquery/client/TableOps.scala

```diff
@@ -158,40 +145,55 @@ private[client] final class TableOps(client: Client) {
     exists(bq.BigQueryHelpers.parseTableSpec(tableSpec))
 
   /** Write rows to a table. */
-  def writeRows(table: TableReference,
+  def writeRows(tableReference: TableReference,
                 rows: List[TableRow],
                 schema: TableSchema,
                 writeDisposition: WriteDisposition,
-                createDisposition: CreateDisposition): Unit = {
-    val options = PipelineOptionsFactory
-      .create()
-      .as(classOf[BigQueryOptions])
-    options.setProject(client.project)
-    options.setGcpCredential(client.credentials)
-    try {
-      val service = new bq.BigQueryServicesWrapper(options)
-      if (createDisposition == CREATE_IF_NEEDED) {
-        service.createTable(new Table().setTableReference(table).setSchema(schema))
-      }
-      service.insertAll(table, rows.asJava)
-    } finally {
-      Option(options.as(classOf[GcsOptions]).getExecutorService)
-        .foreach(_.shutdown())
+                createDisposition: CreateDisposition): Unit = withBigQueryServices { service =>
+    val table = new Table().setTableReference(tableReference).setSchema(schema)
+    if (createDisposition == CreateDisposition.CREATE_IF_NEEDED) {
+      service.createTable(table)
+    }
+
+    writeDisposition match {
+      case WriteDisposition.WRITE_TRUNCATE =>
+        delete(tableReference)
+        service.createTable(table)
+      case WriteDisposition.WRITE_EMPTY =>
+        require(service.isTableEmpty(tableReference))
```

Comment:
I don't particularly like this, but this `writeRows` method is expected to throw 😞